File: backend/models/ChatMessage.js
const mongoose = require("mongoose");

const chatMessageSchema = new mongoose.Schema(
  {
    roomId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "ChatRoom",
      required: true
    },

    senderId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true
    },

    content: {
      type: String,
      required: true
    },

    status: {
      type: String,
      enum: ["sent", "delivered", "seen"],
      default: "delivered"
    }
  },
  { timestamps: true }
);

module.exports = mongoose.model("ChatMessage", chatMessageSchema);

File: lib/src/core/models/chat_message.dart
class ChatMessage {
  final String id;
  final String roomId;
  final String senderId;
  final String content;
  final String status;
  final DateTime createdAt;

  ChatMessage({
    required this.id,
    required this.roomId,
    required this.senderId,
    required this.content,
    required this.status,
    required this.createdAt,
  });

  factory ChatMessage.fromJson(Map<String, dynamic> json) {
    return ChatMessage(
      id: json["_id"] ?? "",
      roomId: json["roomId"] ?? "",
      senderId: json["senderId"] ?? "",
      content: json["content"] ?? "",
      status: json["status"] ?? "sent",
      createdAt: DateTime.parse(json["createdAt"]),
    );
  }
}
